### ISSUE TITLE
PowerDNS bind zone parsing

### ISSUE THREAD

**Issue by EricVeldhuyzen** [PowerDNS/pdns#3568]
We use the Net::DNS perl module to format DNS records in a consistent way. This results for TXT records with long values in something like this:

```
bla       IN      TXT     (
          "v=spf1 include:spf.protection.outlook.com include:mail.foobar.nl -all"
          )
```

When Power DNS read a zonefile with these kind of records it doesn't produce an error (that I can see) but when an AXFR comes for a zone with such a record we see in the log:

```
Mar 14 12:54:12 dnssec1 pdns[2723]: Exception: Parsing record content: Data
field in DNS should start with quote (") at position 3 of '" "v=spf1
include:spf.protection.outlook.com include:mail.foobar.nl -all" "'
Mar 14 12:54:12 dnssec1 pdns[2723]: TCP Connection Thread died because of STL
error: Parsing record content: Data field in DNS should start with quote (") at
position 3 of '" "v=spf1 include:spf.protection.outlook.com
include:mail.foobar.nl -all" "'
```

It seems that the AXFR is not completely failing, only the records that are failing are missing after the AXFR, other records are present and updated, including the SOA (with the serial).

In some other cases the TXT record is read and returned, but after AXFR (from the hidden master) to our public masters we notice extra spaces inside the TXT records between the quote characters. For example, a string like

```
@   IN TXT (
   "LongBase64EncodedString"
   )
```

would become

```
@   IN TXT " LongBase64EncodedString "
```

We use pdns-static 3.4.7-1 amd64, downloaded from downloads.powerdns.com on Debian Wheezy.

**Comment by nlyan** [comment]
Does your hidden master return correct records when queried with dig?

**Comment by Habbie** [comment]
@nlyan no, it SERVFAILs then

**Comment by EricVeldhuyzen** [comment]
Hi, I just tested the fix in the git master release. The problem is solved on this git version. Would you please be so kind to backport this also to the 3.4.* stable branch?

**Comment by Habbie** [comment]
Reopened for backporting.

**Comment by pieterlexis** [comment]
fixed in #3769
